utils/doclint/check_public_api/missingDocs.js

```diff
@@ -0,0 +1,175 @@
+ * limitations under the License.
+ */
+
+const mdBuilder = require('./MDBuilder');
+const Message = require('../Message');
+const ts = require('typescript');
+const EventEmitter = require('events');
+const Documentation = require('./Documentation');
+
+/**
+ * @return {!Array<!Message>}
+ */
+module.exports = function lint(api, jsSources, apiFileName) {
+  const documentation = mdBuilder(api, true).documentation;
+  const apiMethods = listMethods(jsSources, apiFileName);
+  const errors = [];
+  for (const [className, methods] of apiMethods) {
+    const docClass = documentation.classes.get(className);
+    if (!docClass) {
+      errors.push(Message.error(`Missing documentation for "${className}"`));
+      continue;
+    }
+    for (const [methodName, params] of methods) {
+      const member = docClass.members.get(methodName);
+      if (!member) {
+        errors.push(Message.error(`Missing documentation for "${className}.${methodName}"`));
+        continue;
+      }
+      const memberParams = paramsForMember(member);
+      for (const paramName of params) {
+        if (!memberParams.has(paramName))
+          errors.push(Message.error(`Missing documentation for "${className}.${methodName}.${paramName}"`));
+      }
+    }
+  }
+  for (const cls of documentation.classesArray) {
+    const methods = apiMethods.get(cls.name);
+    if (!methods) {
+      errors.push(Message.error(`Documented "${cls.name}" not found in sources`));
+      continue;
+    }
+    for (const member of cls.membersArray) {
+      if (member.kind === 'event')
+        continue;
+      const params = methods.get(member.name);
+      if (!params) {
+        errors.push(Message.error(`Documented "${cls.name}.${member.name}" not found is sources`));
+        continue;
+      }
+      const memberParams = paramsForMember(member);
+      for (const paramName of memberParams) {
+        if (!params.has(paramName))
+          errors.push(Message.error(`Documented "${cls.name}.${member.name}.${paramName}" not found is sources`));
+      }
+    }
+  }
+  return errors;
+};
+
+/**
+ * @param {!Documentation.Member} member
+ */
+function paramsForMember(member) {
+  if (member.kind !== 'method')
+    return [];
+  const paramNames = new Set(member.argsArray.map(a => a.name));
+  if (member.options)
+    paramNames.add('options');
+  return paramNames;
+}
+
+/**
+ * @param {!Array<!import('../Source')>} sources
+ */
+function listMethods(sources, apiFileName) {
+  const program = ts.createProgram({
+    options: {
+      allowJs: true,
+      target: ts.ScriptTarget.ESNext,
+      strict: true
+    },
+    rootNames: sources.map(source => source.filePath())
+  });
+  const checker = program.getTypeChecker();
+  const apiClassNames = new Set();
+  const apiMethods = new Map();
+  const apiSource = program.getSourceFiles().find(f => f.fileName === apiFileName);
+
+  /**
+   * @param {ts.Type} type
+   */
+  function signatureForType(type) {
```

Comment:
JSBuilder lives!

Reply:
I was debating whether to add this check for missing parameters, and decided it only costs us this one JSBuilder function 😄

Reply:
The typescript compiler api isn't quite stable, so this function has actually broken once or twice. But I think we can afford it.